### ISSUE TITLE
Error building opencog/ros-noetic-opencog docker image

### ISSUE THREAD

**Issue by ivan-karlovich** [opencog/docker#182]
### Steps to reproduce
1. Clone opencog docker repo (the latest cloned commit at the time of checking is https://github.com/opencog/docker/commit/2369f9a0e53c7201f4a655e417fd8d780115169c): 
`$ git clone https://github.com/opencog/docker.git`
2. Navigate to `noetic/ros-base` folder.
`$ cd docker/noetic/ros-base/`
3. Run build.sh:
`$ ./build.sh`
4. Test container:
```
$ ./run.sh
# roscore
...
ros_comm version 1.15.15
...
```
5. Navigate to ros-opencog folder:
`$ cd ../ros-opencog/`
6. Run build.sh:
`$ ./build.sh`
7. Build fails at step 40/49:
```
Step 40/49 : RUN (mkdir /opencog/lg-atomese/build; cd /opencog/lg-atomese/build;     cmake ..; make -j12; make install)
 ---> Running in ae05bdd92f11
CMake Deprecation Warning at CMakeLists.txt:19 (CMAKE_POLICY):
  The OLD behavior for policy CMP0037 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type: Release
-- CogUtil found.
-- AtomSpace found.
-- Found PythonInterp: /usr/bin/python (found version "2.7.18") 
-- link-grammar( >= 5.6.2) was not found. Make sure LINK_GRAMMAR_LIBRARY, LINK_GRAMMAR_INCLUDE_DIR and LINK_GRAMMAR_DATA_DIR are set.
-- Link Grammar missing: needed for NLP.
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found UUID: /usr/lib/x86_64-linux-gnu/libuuid.so  
-- Guile (2.2.7 >= 2.2.2) was found.
-- Looking for secure_getenv
-- Looking for secure_getenv - not found
-- Found Python3Interp: /usr/bin/python3  
-- Python 3.8.10 interpreter found.
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found version "3.8.10") 
-- Python 3.8.10 libraries found.
-- Found PythonInterp: /usr/bin/python3.8 (found version "3.8.10") 
-- Found Cython: /usr/bin/cython (Required is at least version "0.24.0") 
-- Cython ( 0.29.1 >= 0.24.0) found.
-- Python destination dir found: /usr/local/lib/python3.8/dist-packages
-- Python install dir: /usr/local/lib/python3.8/dist-packages/opencog
-- Using nosetests executable /usr/bin/nosetests3
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 

Building for Ubuntu 20.04.5 LTS

The following components will be built:
-----------------------------------------------
   Python bindings  - Python (cython) bindings.
   Unit tests       - Unit tests.

The following components WILL NOT be built:
-----------------------------------------------
   LG-Atomese       - Atomese for Link Grammar.
   Doxygen          - Code documentation.

-- Configuring incomplete, errors occurred!
See also "/opencog/lg-atomese/build/CMakeFiles/CMakeOutput.log".
See also "/opencog/lg-atomese/build/CMakeFiles/CMakeError.log".
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LINK_GRAMMAR_LIBRARY (ADVANCED)
    linked by target "lg-dict-entry" in directory /opencog/lg-atomese/opencog/nlp/lg-dict
    linked by target "lg-parse" in directory /opencog/lg-atomese/opencog/nlp/lg-parse

make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.
The command '/bin/sh -c (mkdir /opencog/lg-atomese/build; cd /opencog/lg-atomese/build;     cmake ..; make -j12; make install)' returned a non-zero code: 2

```

### Notes
- Any help in resolution or suggestions in what direction to look to get it fixed are very much appreciated.

**Comment by linas** [comment]
Hi @ivan-karlovich 

If you read through the error messages (in red) you would have seen that `pcre2` was missing. So I fixed it to install `libpcre2-dev`.  (Reading error messages is a good skill to develop...)  I also did a whole bunch of other cleanup, modernizing the script. If you do a `git pull` you'll get the latest, and it should build cleanly.

Thanks for the bug report.  I'm closing this; if there are others, open a new report.
